src/connections.rs

```diff
@@ -288,19 +311,24 @@ async fn read_on_bi_streams(
                         )
                         .await
                         {
-                            error!("Failed to handle Endpoint verification request for peer {:?} with error: {:?}", peer_addr, error);
+                            warn!("Failed to handle Endpoint verification request for peer {:?} with: {:?}", peer_addr, error);
+
+                            return Err(error);
                         }
                     }
                     Ok(msg) => {
-                        error!(
+                        warn!(
                             "Unexpected message type from peer {:?}: {:?}",
                             peer_addr, msg
                         );
                     }
-                    Err(Error::StreamRead(quinn::ReadExactError::FinishedEarly)) => break,
+                    Err(Error::StreamRead(quinn::ReadExactError::FinishedEarly)) => {
+                        warn!("Stream finished early");
+                        break;
+                    }
                     Err(err) => {
-                        error!(
-                            "Failed reading from a bi-stream for peer {:?} with error: {:?}",
+                        warn!(
+                            "Failed reading from a bi-stream for peer {:?} with: {:?}",
                             peer_addr, err
                         );
                         break;
```

Comment:
As I can figure, it's a question of "can this connection still be used". If so, we do not error.

Reply:
So we were breaking before here, so that's not a change. 

As I understand, here we've received the bytes and it's the processing that fails (some message fails), but the connection is still valid, so we don't want to err out and cause a disconnect.